File: app/src/androidTest/java/FilterTest.java
import android.content.Intent;

import net.ducksmanager.whattheduck.CountryList;
import net.ducksmanager.whattheduck.R;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;

import androidx.test.ext.junit.runners.AndroidJUnit4;
import androidx.test.filters.LargeTest;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.action.ViewActions.closeSoftKeyboard;
import static androidx.test.espresso.action.ViewActions.typeText;
import static androidx.test.espresso.assertion.ViewAssertions.doesNotExist;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static androidx.test.espresso.matcher.ViewMatchers.withText;
import static org.hamcrest.Matchers.allOf;

@RunWith(AndroidJUnit4.class)
@LargeTest
public class FilterTest extends WtdTest {

    public FilterTest() {
        super();
    }

    @Before
    public void login() {
        loginActivityRule.launchActivity(new Intent());
        login(DownloadHandlerMock.TEST_USER, DownloadHandlerMock.TEST_PASS);
    }

    @Test
    public void testShowAddIssueDialog() {
        assertCurrentActivityIsInstanceOf(CountryList.class, true);
        onView(allOf(withId(R.id.addToCollectionBySelectionButton), forceFloatingActionButtonsVisible()))
            .perform(click());

        assertCurrentActivityIsInstanceOf(CountryList.class, true);

        onView(withId(R.id.filter))
            .perform(typeText("Fr"), closeSoftKeyboard());

        onView(withText("France"));
        onView(withText("Italy")).check(doesNotExist());


    }
}

File: app/src/androidTest/java/FilterTest.kt

import android.content.Intent
import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions
import androidx.test.espresso.assertion.ViewAssertions
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.filters.LargeTest
import net.ducksmanager.activity.CountryList
import net.ducksmanager.whattheduck.R
import org.junit.Before
import org.junit.Test
import org.junit.runner.RunWith

@RunWith(AndroidJUnit4::class)
@LargeTest
class FilterTest : WtdTest() {
    @Before
    fun login() {
        loginActivityRule.launchActivity(Intent())
        login(DownloadHandlerMock.TEST_USER, DownloadHandlerMock.TEST_PASS)
    }

    @Test
    fun testShowAddIssueDialog() {
        assertCurrentActivityIsInstanceOf(CountryList::class.java, true)

        clickOnActionButton(R.id.addToCollectionBySelectionButton)
        assertCurrentActivityIsInstanceOf(CountryList::class.java, true)

        onView(withId(R.id.filter))
            .perform(ViewActions.typeText("Fr"), ViewActions.closeSoftKeyboard())

        onView(withText("France"))

        onView(withText("Italy")).check(ViewAssertions.doesNotExist())
    }
}